run_dir/static/js/data_del_plot.js

```diff
@@ -0,0 +1,385 @@
+                step: true,
+                name: series_name,
+                color: view_color,
+                data: [],
+            };
+            series.length += 1; 
+        }else if (view_type == 'cumulative'){
+            cumul_xaxis = "Total filesize delivered";
+            dp = {
+                y: fs_sum,
+                data: fs_sum
+            };
+            series[series_name].data.push(dp);
+            categories.push(cumul_xaxis);
+        }else{ 
+            dp = {
+                y: data[d][key],
+                name: project_name + ': ' + "<i>" + date_del + "</i>",
+                ownURL: bioinfo_link
+            };
+            series[series_name].data.push(dp);
+            categories.push(date_del);
+      }
+      var proper_series = []
+      for (s in series) {
+          proper_series.push(series[s]);
+      }
+    }
+  return [categories, proper_series];
+}
+
+function get_plot_data(key, name, view_type, search_string="", filter_inst_type){
+    $.getJSON('/api/v1/proj_staged/'+search_string, function(data) {
+        window.current_plot_data=data;
+        update_color_schemes(data);
+        make_plot(key, name, view_type, filter_inst_type);
+    });
+}
+//Since we have been confused by the coloring in the past, I set a few default colors to re-use for simplicity.
+function view_by_cumul(series_name){
+    return current_color_schemes[1].hex();
+}
+function view_by_chem(series_name){
+    if (series_name == "NovaSeq"){
+      return current_color_schemes[0].hex();
+    }else if (series_name == "MiSeq"){
+      return current_color_schemes[1].hex();
+    }else if (series_name == "NextSeq"){
+      return current_color_schemes[2].hex();
+    }else{
+      return "#c3c3c3";
+    }
+}
+function view_by_del(series_name){
+    if (series_name == 'HDD'){
+      return current_color_schemes[0].hex();
+    }else if (series_name == 'GRUS'){
+      return current_color_schemes[2].hex();
+    }else{
+      return "#c3c3c3";
+    }
+}
+function view_by_app(series_name){
+    if (series_name == 'RNA-Seq'){
+      return current_color_schemes[0].hex();
+    }else if (series_name == 'WG-reseq'){
+      return current_color_schemes[1].hex();
+    }else if (series_name == 'Target-reseq'){
+      return current_color_schemes[2].hex();
+    }else if (series_name == 'Metagenomics'){
+      return current_color_schemes[3].hex();
+    }else if (series_name == 'de novo'){
+      return current_color_schemes[4].hex();
+    }else if (series_name == 'Epigenetics'){
+      return current_color_schemes[5].hex();
+    }else{
+      return "#c3c3c3";
+    } 
+} 
+function view_by_ngi(series_name){
+    if (series_name == 'Production'){
+      return current_color_schemes[5].hex();
+    }else if (series_name == 'Application'){
+      return current_color_schemes[1].hex();
+    }else{
+      return "#c3c3c3";
+    }
+}
+function view_by_sam(series_name){
+    if (series_name == 'Finished Library'){
```

Comment:
This should probably be a javascript switch statement instead of a if else: https://www.w3schools.com/js/js_switch.asp

Reply:
Huh I've never heard of the `switch` statement, interesting! OK, how's this, I don't know how pretty it turned out lol

Reply:
I think it looks much better! :) Making it one big one instead of the smaller ones wasn't necessary but I don't mind it either.